internals/aws/service_creator.go

```diff
@@ -31,28 +31,36 @@ type CredentialCreator struct {
 
 // NewCredentialCreator returns a CredentialCreator that uses the provided AWS KMS key and IAM role to create a new credential.
 // The AWS credential is configured with the optionally provided aws.Config.
-func NewCredentialCreator(keyID, role string, cfgs ...*aws.Config) (*CredentialCreator, error) {
+func NewCredentialCreator(keyID, role string, cfgs ...*aws.Config) (*CredentialCreator, map[string]string, error) {
```

Comment:
It looks like this signature is not compatible with the changes made in #117

Reply:
We can evaluate the role and key in `CreateAWS` instead.